### PR TITLE
test: Fix flow validation retries

### PR DESCRIPTION
**Pull request by jrajahalme**
Retry should be done when Retry() return nil, not when it fails.

Set the warning interval so that we get at least one log for
retries. The default was 10 seconds which was equal to the whole flow
retry timeout, so we never saw any logs for retries.

Fixes: #158
Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>

### CODE REVIEW

**Comment by michi-covalent** [comment]
@jrajahalme is https://github.com/cilium/cilium-cli/pull/301/commits/7e0a3b2b66b17a0fe7f309978ca1afbb4dd514ec meant to be include in the PR or are you still debugging something?

File: connectivity/check/check.go
Comment (pchaigno) [inline connectivity/check/check.go:172]: What does `XXX` mean here?

**Review by pchaigno — APPROVED** [review]
I've read the code, but I'm very unfamiliar with that codebase. It seems the current version of the PR also doesn't require my review anymore anyway. I've therefore removed the stale review requests.

File: connectivity/check/check.go
Comment (jrajahalme) [reply connectivity/check/check.go:172]: Addressed this by expanding the function comment as follows:
```
+// append adds any flows in 'from' to 'f', if the flow's timestamp is
+// later than the last one in 'f', returns the combined set.
```
This logic would fail to add a flow if 'from' contains flows with the same timestamp as the last in 'f'. My understanding is that when 'from' is a fresh batch of flows received from Hubble this should not happen.

**Comment by jrajahalme** [comment]
> @jrajahalme is [7e0a3b2](https://github.com/cilium/cilium-cli/commit/7e0a3b2b66b17a0fe7f309978ca1afbb4dd514ec) meant to be include in the PR or are you still debugging something?

No, that was debugging, not in the PR anymore :-)

**Comment by jrajahalme** [comment]
The failing multicluster test is fixed in #302, not related to this PR.

**Comment by pchaigno** [comment]
Aren't we still missing a review to mark `ready-to-merge`? Especially considering I'm not very confident in my review.

**Comment by michi-covalent** [comment]
removed `ready-to-merge`. let's wait for review from cilium/policy. the change lgtm though.

**Review by kkourt — APPROVED** [review]

**Pull request merged**
